include/exec/sequence_senders.hpp

```diff
@@ -0,0 +1,777 @@
+         } -> __sequence_sender::is_valid_next_completions<stdexec::__decay_t<_Receiver>>;
+       };
+
+  template <class _Receiver, class _Sender>
+  concept sequence_receiver_from = sequence_receiver_of<
+    _Receiver,
+    stdexec::completion_signatures_of_t<_Sender, stdexec::env_of_t<_Receiver>>>;
+
+  namespace __sequence_sender {
+    struct sequence_connect_t;
+
+    template <class _Sender, class _Receiver>
+    concept __sequence_connectable_with_tag_invoke =
+      receiver<_Receiver> &&                        //
+      sender_in<_Sender, env_of_t<_Receiver>> &&    //
+      sequence_receiver_from<_Receiver, _Sender> && //
+      tag_invocable<sequence_connect_t, _Sender, _Receiver>;
+
+    struct sequence_connect_t {
+      template <class _Sender, class _Receiver>
+        requires __sequence_connectable_with_tag_invoke<_Sender, _Receiver>
+      auto operator()(_Sender&& __sender, _Receiver&& __rcvr) const
+        noexcept(nothrow_tag_invocable<sequence_connect_t, _Sender, _Receiver>)
+          -> tag_invoke_result_t<sequence_connect_t, _Sender, _Receiver> {
+        static_assert(
+          operation_state<tag_invoke_result_t<sequence_connect_t, _Sender, _Receiver>>,
+          "exec::sequence_connect(sender, receiver) must return a type that "
+          "satisfies the operation_state concept");
+        return tag_invoke(*this, (_Sender&&) __sender, (_Receiver&&) __rcvr);
+      }
+    };
+
+    template <class _Sender, class _Receiver>
+    using sequence_connect_result_t = __call_result_t<sequence_connect_t, _Sender, _Receiver>;
+  } // namespace __sequence_sender
+
+  using __sequence_sender::sequence_connect_t;
+  inline constexpr sequence_connect_t sequence_connect;
+
+  using __sequence_sender::sequence_connect_result_t;
+
+  template <class _Sender, class _Receiver>
+  concept sequence_sender_to =
+    stdexec::sender_in<_Sender, stdexec::env_of_t<_Receiver>>
+    && sequence_receiver_from<_Receiver, _Sender>
+    && requires(_Sender&& __sndr, _Receiver&& __rcvr) {
+         { sequence_connect((_Sender&&) __sndr, (_Receiver&&) __rcvr) } -> stdexec::operation_state;
+       };
+
+  namespace __repeat_each {
+    using namespace stdexec;
+
+    // Takes a sender and creates a sequence sender by repeating the sender as item to the set_next
+    // operation.
+    template <class _SourceSenderId, class _ReceiverId>
+    struct __operation {
+      using _SourceSender = stdexec::__t<_SourceSenderId>;
+      using _Receiver = stdexec::__t<_ReceiverId>;
+      using __next_sender_t = __call_result_t<set_next_t, _Receiver&, _SourceSender>;
+      using __next_on_scheduler_sender =
+        __call_result_t<on_t, exec::trampoline_scheduler, __next_sender_t&&>;
+
+      struct __t;
+
+      struct __receiver {
+        __t* __op_;
+
+        template <class... _Args>
+        friend void tag_invoke(set_value_t, __receiver&& __self, _Args&&...) noexcept {
+          __self.__op_->repeat();
+        }
```

Comment:
It seems a little strange to me that we would just drop these results on the floor. @kirkshoop thoughts?

Reply:
I've noodled over this question and this turned out to be deeper than I thought.

1. I think you are correct. Its iffy.
2. `completion_signatures` for sequence senders describe the signatures that are passed *to* the `set_next` CPO of a sequence receiver. I believe we should additionally require that the sender *returned by* `set_next` should be a single value typed `sender_of<set_value_t()>`. This would be an additional requirement to `set_next`.